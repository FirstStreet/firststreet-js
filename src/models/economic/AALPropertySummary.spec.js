const { isArray } = require('lodash');
const Sut = require('./AALPropertySummary.js');

const mock = require('../../__mocks__/aalPropertySummary.json');

describe('AAL property summary', () => {
  it('should construct object', () => {
    const result = new Sut(mock);
    expect(result.raw).toMatchSnapshot();
    expect(result.raw.depthLoss).toBe(mock.depthLoss);
    expect(result.raw.annualLoss).toBe(mock.annualLoss);
    expect(isArray(result.depthLoss)).toBe(true);
    expect(result.depthLoss.length).toBe(3);
    expect(isArray(result.annualLoss)).toBe(true);
    expect(result.annualLoss.length).toBe(2);
    expect(result.depthLoss[0].constructor.name).toBe('AALDepthLoss');
    expect(result.annualLoss[0].constructor.name).toBe('AALAnnualLoss');
    expect(result.depthLoss[0].data).toEqual(25081);
  });

  it('should construct object without child nodes', () => {
    const result = new Sut({});
    expect(result.raw).toMatchSnapshot();
    expect(result.raw.depthLoss).toBeUndefined();
    expect(result.raw.annualLoss).toBeUndefined();
    expect(result.depthLoss).toBeUndefined();
    expect(result.annualLoss).toBeUndefined();
  });

  it('should construct object with empty child nodes', () => {
    const result = new Sut({ depthLoss: [], annualLoss: [] });
    expect(result.raw).toMatchSnapshot();
    expect(result.raw.depthLoss).toEqual([]);
    expect(result.raw.annualLoss).toEqual([]);
    expect(result.depthLoss).toEqual([]);
    expect(result.annualLoss).toEqual([]);
  });
});
