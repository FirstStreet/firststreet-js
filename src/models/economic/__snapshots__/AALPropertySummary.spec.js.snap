// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AAL property summary should construct object 1`] = `
Object {
  "annualLoss": Array [
    Object {
      "data": Object {
        "high": 9244,
        "low": 7678,
        "mid": 8469,
      },
      "year": 2020,
    },
    Object {
      "data": Object {
        "high": 9950,
        "low": 8366,
        "mid": 9106,
      },
      "year": 2050,
    },
  ],
  "depthLoss": Array [
    Object {
      "data": 25081,
      "depth": 3,
    },
    Object {
      "data": 28472,
      "depth": 15,
    },
    Object {
      "data": 32827,
      "depth": 31,
    },
  ],
}
`;

exports[`AAL property summary should construct object with empty child nodes 1`] = `
Object {
  "annualLoss": Array [],
  "depthLoss": Array [],
}
`;

exports[`AAL property summary should construct object without child nodes 1`] = `Object {}`;
